### PR TITLE
Add enableSuggestions flag to TextField and TextFormField

### LINKED ISSUE #22828: TextField: "autocorrect: false" still shows suggestions in keyboard
Flutter version: Very recent master build

Happens in the Android Emulator (with Google Keyboard) and on a real Android device (with Sony keyboard). I'm not sure about iOS.

Code to reproduce:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_playground/text_area.dart';
import 'package:path/path.dart' as path;

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Playground',
      home: AutoCorrectTest(),
    );
  }
}

class AutoCorrectTest extends StatefulWidget {
  @override
  _AutoCorrectTestState createState() => _AutoCorrectTestState();
}

class _AutoCorrectTestState extends State<AutoCorrectTest> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('TextField with autocorrect: false'),
      ),
      body: Padding(
        padding: EdgeInsets.all(16.0),
        child: TextField(
          autocorrect: false
        )
      ),
    );
  }
}
```

```
[√] Flutter (Channel master, v0.9.7-pre.61, on Microsoft Windows [Version 10.0.17134.285], locale de-DE)
    • Flutter version 0.9.7-pre.61 at C:\flutter
    • Framework revision 2d81adf74c (3 days ago), 2018-10-05 22:29:37 -0700
    • Engine revision 572fa5646a
    • Dart version 2.1.0-dev.6.0.flutter-c6254163dc

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at F:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = F:\Android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 29.0.2
    • Dart plugin version 181.5616
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[!] VS Code, 64-bit edition
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)
```

### LINKED ISSUE #41758: TextField autocorrect=false bug
Filing new bug to draw attention to:

https://github.com/flutter/flutter/issues/22828#issuecomment-520174825

Flutter is not setting the necessary InputType flags when autocorrect is set to false.

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Currently there is no direct control over the `TYPE_TEXT_FLAG_NO_SUGGESTIONS` flag in Android.  This PR adds a `enableSuggestions` flag to all types of text fields to control this Android flag.

## Related Issues

Closes https://github.com/flutter/flutter/issues/22828
Closes https://github.com/flutter/flutter/issues/41758
Depends on engine PR: https://github.com/flutter/engine/pull/13099

## Tests

Testing that the property is set and that it goes to the engine correctly.  I don't think there's a more thorough way to test that.

### CODE REVIEW

**Comment by Blindjoedeath** [comment]
Dude, you are doing the great job!!!! God bless your soul..
This is really important for my recent project.

**Comment by justinmc** [comment]
No problem! 🎉 I'm trying to figure out some tests for this and then it should be ready for code review.

**Comment by justinmc** [comment]
Is `noSuggestions` a good name?  Just mirroring the Android name (and the negative).  We could also go with something like `keyboardSuggestions` or `allowKeyboardSuggestions` and invert it.  I'm not strongly opinionated.

**Comment by jonahwilliams** [comment]
I think it would be better to state it in the positive, rather than be consistent with a (IMO) bad API.

**Comment by justinmc** [comment]
Thanks @jonahwilliams, I'm on board with that.  I'll plan to change it to `enableKeyboardSuggestions` unless anyone else disagrees?

**Comment by Blindjoedeath** [comment]
@justinmc  How is your work going? Is it possible to get your commits in near future?:)

**Comment by justinmc** [comment]
Sorry for the delay!  I'm changing the name to `enableSuggestions`.  The engine PR should be able to be merged in the next few hours, and then this PR can follow.

File: packages/flutter/lib/src/services/text_input.dart
Comment (HansMuller) [inline packages/flutter/lib/src/services/text_input.dart]: Could we provide a brief explanation of what "suggestions" actually are?

File: packages/flutter/lib/src/services/text_input.dart
Comment (HansMuller) [inline packages/flutter/lib/src/services/text_input.dart]: Maybe link "Android" to the native doc that describes the feature we depend on.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by TutOsirisOm** [comment]
What's the status on this?

**Comment by justinmc** [comment]
Just waiting for the build to be green so I can merge this.

I just verified with the engine on master that this branch works and passes the flag correctly, so everything is good to go.

**Comment by justinmc** [comment]
What version are you using?  I think this is in v1.16.3 and above.

**Comment by YaseenAbdullah** [comment]
> What version are you using? I think this is in v1.16.3 and above.

Latest version 1.17.0, i updated it last night

**Comment by justinmc** [comment]
Hmm thanks for checking.  Does the flag `TYPE_TEXT_FLAG_NO_SUGGESTIONS` solve your problem in native Android?  It looks like you're using the default Gboard keyboard, and I was seeing it ignore that flag even in a native app when I tested this.

**Comment by YaseenAbdullah** [comment]
> Hmm thanks for checking. Does the flag `TYPE_TEXT_FLAG_NO_SUGGESTIONS` solve your problem in native Android? It looks like you're using the default Gboard keyboard, and I was seeing it ignore that flag even in a native app when I tested this.

I didn't tested it in native android but i use SwiftKey in my physical device it's the same

**Comment by YaseenAbdullah** [comment]
any hint how to make it work? I really need it in my project

**Comment by justinmc** [comment]
See this comment: https://github.com/flutter/flutter/issues/22828#issuecomment-566269059

If your problem is the same as that, then it might not be possible to achieve even on native Android.  If Flutter is missing something that native has we can look into adding support, though.

**Comment by NickalasB** [comment]
FWIW setting `enableSuggestions:false` doesn't work for me either. I'm using Pixel 3/Android 11

```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.7 19H114 darwin-x64, locale en-US)
```

This issue https://github.com/flutter/flutter/issues/22828 outlines the issue but was closed after this was merged

**Pull request merged**
